Scan-build warnings

After the intense leak search, I decided to run Clang's `scan-build` on LibreSplit, with the following warnings that might need some attention (or might not).

Output has been cut for ease of reading.

```
../../../src/lasr/maps/maps.c:45:27: warning: Access to field 'next' results in a dereference of a null pointer (loaded from variable 'current') [core.NullDereference]
   45 |             current->next = new_block;
      |             ~~~~~~~       ^
1 warning generated.
...
../../../src/gui/theming.c:153:13: warning: Value stored to 'error' is never read [deadcode.DeadStores]
  153 |             error = true;
      |             ^       ~~~~
1 warning generated.
...
../../../src/gui/component/prev-segment.c:142:9: warning: Value stored to 'prev' is never read [deadcode.DeadStores]
  142 |         prev = timer->curr_split - 1;
      |         ^      ~~~~~~~~~~~~~~~~~~~~~
1 warning generated.
...
../../../src/keybinds/bind.c:584:13: warning: Value stored to 'iter' during its initialization is never read [deadcode.DeadStores]
  584 |     GSList* iter = bindings;
      |             ^~~~   ~~~~~~~~
1 warning generated.
...
../../../src/timer.c:458:9: warning: Potential leak of memory pointed to by 'game' [unix.Malloc]
  458 |     if (json) {
      |         ^~~~
../../../src/timer.c:822:12: warning: Potential leak of memory pointed to by 'timer' [unix.Malloc]
  822 |     return error;
      |            ^~~~~
2 warnings generated.
```

Some may be fixed by the open PRs but I want to keep the progress of this analysis and respective solution here.

--- a/src/gui/component/prev-segment.c
+++ b/src/gui/component/prev-segment.c
@@ -139,7 +139,6 @@ static void prev_segment_draw(LSComponent* self_, const ls_game* game,
         ls_delta_string(str, timer->segment_deltas[curr]);
         gtk_label_set_text(GTK_LABEL(self->previous_segment), str);
     } else if (curr) {
-        prev = timer->curr_split - 1;
         // Previous segment
         if (timer->curr_split) {
             prev = timer->curr_split - 1;

--- a/src/gui/theming.c
+++ b/src/gui/theming.c
@@ -150,7 +150,6 @@ void ls_app_load_theme_with_fallback(LSAppWindow* win, const char* name, const c
             (gssize)fallback_css_data_len(), &gerror);
         if (gerror != NULL) {
             g_printerr("Error loading default theme CSS: %s\n", gerror->message);
-            error = true;
             g_error_free(gerror);
             gerror = NULL;
         }

--- a/src/keybinds/bind.c
+++ b/src/keybinds/bind.c
@@ -548,9 +548,8 @@ keybinder_bind_full(const char* keystring,
  */
 void keybinder_unbind(const char* keystring, KeybinderHandler handler)
 {
-    GSList* iter;
 
-    for (iter = bindings; iter != NULL; iter = iter->next) {
+    for (GSList* iter = bindings; iter != NULL; iter = iter->next) {
         struct Binding* binding = iter->data;
 
         if (strcmp(keystring, binding->keystring) != 0 || handler != binding->handler)
@@ -581,9 +580,8 @@ void keybinder_unbind(const char* keystring, KeybinderHandler handler)
  */
 void keybinder_unbind_all(const char* keystring)
 {
-    GSList* iter;
 
-    for (iter = bindings; iter != NULL; iter = iter->next) {
+    for (GSList* iter = bindings; iter != NULL; iter = iter->next) {
         struct Binding* binding = iter->data;
 
         if (strcmp(keystring, binding->keystring) != 0) {

--- a/src/keybinds/bind.c
+++ b/src/keybinds/bind.c
@@ -581,7 +581,7 @@ void keybinder_unbind(const char* keystring, KeybinderHandler handler)
  */
 void keybinder_unbind_all(const char* keystring)
 {
-    GSList* iter = bindings;
+    GSList* iter;
 
     for (iter = bindings; iter != NULL; iter = iter->next) {
         struct Binding* binding = iter->data;

--- a/src/lasr/auto-splitter.c
+++ b/src/lasr/auto-splitter.c
@@ -209,6 +209,7 @@ bool call_va(lua_State* L, const char* func, const char* sig, ...)
 
             default:
                 printf("invalid option (%c)\n", *(sig - 1));
+                va_end(vl);
                 return false;
         }
         if (*(sig - 1) == '>')
@@ -235,6 +236,7 @@ bool call_va(lua_State* L, const char* func, const char* sig, ...)
                 case 'd': /* double result */
                     if (!lua_isnumber(L, nres)) {
                         printf("function '%s' wrong result type, expected double\n", func);
+                        va_end(vl);
                         return false;
                     }
                     *va_arg(vl, double*) = lua_tonumber(L, nres);
@@ -243,6 +245,7 @@ bool call_va(lua_State* L, const char* func, const char* sig, ...)
                 case 'i': /* int result */
                     if (!lua_isnumber(L, nres)) {
                         printf("function '%s' wrong result type, expected int\n", func);
+                        va_end(vl);
                         return false;
                     }
                     *va_arg(vl, int*) = lua_tointeger(L, nres);
@@ -251,6 +254,7 @@ bool call_va(lua_State* L, const char* func, const char* sig, ...)
                 case 's': /* string result */
                     if (!lua_isstring(L, nres)) {
                         printf("function '%s' wrong result type, expected string\n", func);
+                        va_end(vl);
                         return false;
                     }
                     *va_arg(vl, const char**) = lua_tostring(L, nres);
@@ -259,6 +263,7 @@ bool call_va(lua_State* L, const char* func, const char* sig, ...)
                 case 'b':
                     if (!lua_isboolean(L, nres)) {
                         printf("function '%s' wrong result type, expected boolean\n", func);
+                        va_end(vl);
                         return false;
                     }
                     *va_arg(vl, bool*) = lua_toboolean(L, nres);
@@ -266,6 +271,7 @@ bool call_va(lua_State* L, const char* func, const char* sig, ...)
 
                 default:
                     printf("invalid option (%c)\n", *(sig - 1));
+                    va_end(vl);
                     return false;
             }
             nres++;

--- a/src/lasr/maps/maps.c
+++ b/src/lasr/maps/maps.c
@@ -39,10 +39,14 @@ static void append_entry(ProcessMap e)
         new_block->used = 0;
         new_block->next = NULL;
 
-        if (!head)
+        if (!head) {
             head = new_block;
-        else
-            current->next = new_block;
+        } else {
+            if (current) {
+                // Guard to avoid null-dereferencing
+                current->next = new_block;
+            }
+        }
 
         current = new_block;
     }

--- a/src/timer.c
+++ b/src/timer.c
@@ -4,6 +4,7 @@
  */
 #include "timer.h"
 #include "gui/dialogs.h"
+#include "logging.h"
 #include "settings/utils.h"
 
 #include "lasr/auto-splitter.h"
@@ -287,6 +288,11 @@ int ls_game_create(ls_game** game_ptr, const char* path, char** error_msg)
         error = 1;
         size_t msg_len = snprintf(NULL, 0, "%s (%d:%d)", json_error.text, json_error.line, json_error.column);
         *error_msg = calloc(msg_len + 1, sizeof(char));
+        if (*error_msg == NULL) {
+            LOG_ERR("Cannot allocate memory for error message");
+            error = 1;
+            goto game_create_error;
+        }
         sprintf(*error_msg, "%s (%d:%d)", json_error.text, json_error.line, json_error.column);
         goto game_create_error;
     }

